src/components/PriceWebSocket/PriceWebSocket.ts

```diff
@@ -0,0 +1,29 @@
+import { WebSocketMessage } from "./types";
+
+const PRICE_WS_MSG_EVENT = "message";
+
+const processPriceSocketMsg = function (incomingMsg: string): WebSocketMessage {
+  return JSON.parse(incomingMsg) as WebSocketMessage;
+}
+
+export const createPriceWebSocketConn = function (uri: string): WebSocket {
+  return new WebSocket(uri);
+}
+
+export const closePriceWebSocketConn = function (ws: WebSocket) {
+  return ws.close();
+}
+
+export const initWsConnAndAddEventListener = function (
```

Comment:
Going to break this up, functions should have a singular responsibility,

so InitWebSocket() should be one responsibility

adding EventListener should be another. 


According to the Gang of Four design patterns, functions (or classes) should ideally have a single responsibility. This principle is known as the Single Responsibility Principle (SRP), which is one of the SOLID principles of object-oriented design. The Gang of Four (GoF) design patterns, as outlined in the book "Design Patterns: Elements of Reusable Object-Oriented Software" by Erich Gamma, Richard Helm, Ralph Johnson, and John Vlissides, do not explicitly state this principle, but they do promote the idea of designing classes and objects that have well-defined and focused responsibilities.

The Single Responsibility Principle suggests that a class or function should have only one reason to change. In other words, it should have a single, clearly defined purpose or responsibility. This helps in making your code more maintainable, understandable, and easier to modify in the future. When a class or function has multiple responsibilities, it can become complex, difficult to test, and prone to unintended side effects.

Design patterns often encourage the separation of concerns and the use of small, focused components that can be composed together to achieve more complex functionality. By adhering to the Single Responsibility Principle, you can create code that is more modular and easier to maintain, and you can also use design patterns more effectively to solve specific design problems.

Reply:
Resolved: https://github.com/ChaoticDreamsCorp/shellhacks_crypto_price/pull/2/commits/1d7cfae4c75196fb3b133caec4751428a525dee5